moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -526,13 +544,16 @@ double JogCalcs::velocityScalingFactorForSingularity(const Eigen::VectorXd& comm
       velocity_scale = 1. -
                        (ini_condition - parameters_.lower_singularity_threshold) /
                            (parameters_.hard_stop_singularity_threshold - parameters_.lower_singularity_threshold);
+      status_ = kSingularityDecelerate;
+      ROS_WARN_STREAM_THROTTLE_NAMED(2, LOGNAME, kStatusCodeMap.at(status_));
     }
 
     // Very close to singularity, so halt.
     else if (ini_condition > parameters_.hard_stop_singularity_threshold)
     {
       velocity_scale = 0;
-      ROS_WARN_THROTTLE_NAMED(2, LOGNAME, "Close to a singularity. Halting.");
+      status_ = kSingularityHalt;
```

Comment:
Is there a reason you are using the mutex for reading but not for writing the status?

Reply:
Looking through jog_calcs.cpp, status_ gets written to the shared variables here (L385):

```
  mutex.lock();
  shared_variables.status = status_;
  mutex.unlock();
```

Mutexes aren't used anywhere else. I think that makes sense.